src/braket/emulators/emulator_passes/criteria/gate_criterion.py

```diff
@@ -0,0 +1,68 @@
+from collections.abc import Iterator
+
+from braket.circuits import Circuit
+from braket.circuits.compiler_directives import EndVerbatimBox, StartVerbatimBox
+from braket.circuits.gate import Gate
+from braket.circuits.translations import BRAKET_GATES
+from braket.emulators.emulator_passes.criteria.emulator_criterion import EmulatorCriterion
+
+
+class GateCriterion(EmulatorCriterion):
+    def __init__(self, supported_gates: Iterator[str] = [], native_gates: Iterator[str] = []):
+        """
+        args:
+            native_gates (Iterator[str]): A list of gates supported inside of verbatim mode by
+            the emulator.
+            supported_gates (Iterator[str]): A list of gates supported outside of verbatim mode
+            by the emulator. A gate is a Braket gate name.
+        """
+        if len(supported_gates) == 0 and len(native_gates) == 0:
+            raise ValueError("Supported gate set or native gate set must be provided.")
+
+        try:
+            self._supported_gates = set(BRAKET_GATES[gate.lower()] for gate in supported_gates)
+            self._native_gates = set(BRAKET_GATES[gate.lower()] for gate in native_gates)
+        except KeyError as e:
+            raise ValueError(f"Input {str(e)} is not a valid Braket gate name.")
+
+    def validate(self, circuit: Circuit) -> None:
+        """
+        Checks that all non-verbatim gates used in the circuit are in this criterion's
+        supported gate set and that all verbatim gates used in the circuit are in this
+        criterion's native gate set.
+
+        Args:
+            circuit (Circuit): The Braket circuit whose gates to validate.
+        """
+        idx = 0
+        while idx < len(circuit.instructions):
```

Comment:
High level questions. I thought if there is a verbatim box in the circuit, then all the gates have to be native gates. 
1. Does the following validation satisfies this requirement? I wasn't able to tell yet.
2. Can we simply check if there is a verbatim box in the circuit, and check against either the supported gate set and native gate set accordingly for all the gates?

Reply:
1. If there is a verbatim box in the circuit, all the gates _don't_ have to be native gates - just the ones in the verbatim box. Supported gates outside the verbatim box can still be nativized without any issue!

**_However_**, if there is a verbatim box, then _all qubit references must be to hardware qubits_ (this is checked in `ConnectivityCriterion`). 

2. Because of the comment in response to (1), we need to use both the supported gate set and the native gate set when validating a circuit.